### PR TITLE
Update RxJava adapter to 1.2.0.

### PR DESCRIPTION
**Pull request by JakeWharton**
Use a more idiomatic creation mechanism which eliminates a lot of responsibility for our implementation.

### CODE REVIEW

File: retrofit-adapters/rxjava/src/test/java/retrofit2/adapter/rxjava/ObservableTest.java [hunk 1]
```diff
@@ -86,83 +81,16 @@
 
     RecordingSubscriber<String> subscriber = subscriberRule.createWithInitialRequest(0);
     service.body().unsafeSubscribe(subscriber);
-    assertThat(server.getRequestCount()).isEqualTo(0);
+    assertThat(server.getRequestCount()).isEqualTo(1);
+    subscriber.assertNoEvents();
```
Comment (JakeWharton) [inline retrofit-adapters/rxjava/src/test/java/retrofit2/adapter/rxjava/ObservableTest.java]: This is an unrelated internal behavior change I've been meaning to make. It has no effect on what the downstream consumer of the stream sees in terms of behavior.

File: retrofit-adapters/rxjava/src/main/java/retrofit2/adapter/rxjava/RxJavaCallAdapter.java [hunk 2]
```diff
@@ -46,24 +44,23 @@
   }
 
   @Override public Object adapt(Call<R> call) {
-    OnSubscribe<Response<R>> callFunc = new CallOnSubscribe<>(call);
+    ResponseCallable<R> resultCallable = new ResponseCallable<>(call);
```
Comment (swankjesse) [inline retrofit-adapters/rxjava/src/main/java/retrofit2/adapter/rxjava/RxJavaCallAdapter.java]: I like this new callable-driven approach
